ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTPayload.scala

```diff
@@ -183,9 +189,8 @@ object AuthServiceJWTCodec {
   }
 
   def readStandardTokenPayload(jsValue: JsValue): Option[AuthServiceJWTPayload] = jsValue match {
-    // NOTE: there is the corner-case of a legacy Daml token containing a "sub" field.
-    // We accept that risk.
-    case JsObject(fields) if !fields.contains(oidcNamespace) && fields.contains("sub") =>
+    case JsObject(fields)
+        if fields.contains("sub") && !distinguishingCustomProps.exists(fields.contains) =>
```

Comment:
This is slightly orthogonal but I find our APIs a bit confusing. If I’m just looking for something to decode a token, the first thing I’d try is `AuthServiceJWTCodec.readPayload`. However, that will only read non-standard tokens and to make things more confusing it will interpret a standard token as the legacy non-standard token which leads to rather confusing error messages.

I’d prefer if `readPayload` was the generic one that supports all formats and if we need to expose that API at all, the more specific ones have more specific names.

Reply:
Good point. I'll address that in https://github.com/digital-asset/daml/issues/12049